django/db/models/fields/related.py

```diff
@@ -986,6 +986,9 @@ def formfield(self, **kwargs):
     def save_form_data(self, instance, data):
         if isinstance(data, self.remote_field.model):
             setattr(instance, self.name, data)
+        elif data is None:
```

Comment:
Can you add a comment about this? Maybe add an `if data is None` branch under the existing `else` clause rather than duplicating `setattr(instance, self.attname, data)`?